codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -870,7 +875,8 @@ public void close(ChannelPromise promise) {
 
                     // Only ever send a reset frame if the connection is still alive as otherwise it makes no sense at
                     // all anyway.
-                    if (parent().isActive() && !streamClosedWithoutError && isStreamIdValid(stream().id())) {
+                    if (!outboundClosed && parent().isActive() &&
```

Comment:
IIUC `outboundClosed` maybe set to `true` as a result of detecting the connection is in `HALF_CLOSED_REMOTE` in the flow controller (via `writeComplete`). If this is the case we still may want to send a `RST_STREAM` if we have not yet closed the local side of the stream.

Reply:
@Scottmitch hm... maybe we should just not care about `outboundClosed` here